core/services/keystore/keys/csakey/key_v2.go

```diff
@@ -38,6 +40,14 @@ type KeyV2 struct {
 	Version    int
 }
 
+func (k KeyV2) StaticSizedPublicKey() (sspk credentials.StaticSizedPublicKey) {
+	if len(k.PublicKey) != ed25519.PublicKeySize {
+		panic(fmt.Sprintf("expected ed25519.PublicKey to have len %d but got len %d", ed25519.PublicKeySize, len(k.PublicKey)))
```

Comment:
How many services use CSA keys?  If it's not all of them, could panic be too strong of a reaction?

Reply:
(Someone recently suggested this is only used for FMS, but looking at this code it seems more important than that.)